services/nyx/build_afl.sh

```diff
@@ -57,66 +57,23 @@ mkdir -p /srv/repos
 pushd /srv/repos >/dev/null
 git-clone-rev https://github.com/AFLplusplus/AFLplusplus 78b7e14c73baacf1d88b3c03955e78f5080d17ba
 pushd AFLplusplus >/dev/null
-# WIP 2-byte chunked variant of honggfuzz custom mutator
-retry-curl https://github.com/AFLplusplus/AFLplusplus/commit/1b611bb30c14724f0f2eb9330772d30723ba122c.diff | git apply
-git apply << "EOF"
-diff --git a/custom_mutators/honggfuzz/Makefile b/custom_mutators/honggfuzz/Makefile
-index 5c2fcddb..2dde8ba1 100644
---- a/custom_mutators/honggfuzz/Makefile
-+++ b/custom_mutators/honggfuzz/Makefile
-@@ -1,5 +1,5 @@
-
--CFLAGS = -O3 -funroll-loops -fPIC -Wl,-Bsymbolic
-+CFLAGS = -O3 -funroll-loops -fPIC -fblocks -lBlocksRuntime -Wl,-Bsymbolic
-
- all: honggfuzz-mutator.so honggfuzz-2b-chunked-mutator.so
-
-commit d606e18332b4f919780604b9daf9a3761602b7c5
-Author: Jesse Schwartzentruber <truber@mozilla.com>
-Date:   Fri Jul 14 11:04:04 2023 -0400
-
-    Increase MAP_SIZE for Nyx
 
-diff --git a/include/config.h b/include/config.h
-index 8585041e..6e526717 100644
---- a/include/config.h
-+++ b/include/config.h
-@@ -442,7 +442,7 @@
-    problems with complex programs). You need to recompile the target binary
-    after changing this - otherwise, SEGVs may ensue. */
-
--#define MAP_SIZE_POW2 16
-+#define MAP_SIZE_POW2 23
-
- /* Do not change this unless you really know what you are doing. */
-
-EOF
+# WIP 2-byte chunked variant of honggfuzz custom mutator
+git apply /srv/repos/setup/patches/hongfuzz-2b-chunked.diff
 make -f GNUmakefile afl-fuzz afl-showmap CODE_COVERAGE=1
 pushd custom_mutators/honggfuzz >/dev/null
 make
 popd >/dev/null
+
 pushd nyx_mode >/dev/null
 git submodule init
+
 retry git submodule update --depth 1 --single-branch libnyx
 pushd libnyx >/dev/null
-git apply << "EOF"
-diff --git a/fuzz_runner/src/nyx/qemu_process.rs b/fuzz_runner/src/nyx/qemu_process.rs
-index d062d87..c4ebeea 100644
---- a/fuzz_runner/src/nyx/qemu_process.rs
-+++ b/fuzz_runner/src/nyx/qemu_process.rs
-@@ -97,9 +97,7 @@ impl QemuProcess {
-     pub fn new(params: QemuParams) -> Result<QemuProcess, String> {
-         Self::prepare_redqueen_workdir(&params.workdir, params.qemu_id);
-
--        if params.qemu_id == 0{
--            println!("[!] libnyx: spawning qemu with:\n {}", params.cmd.join(" "));
--        }
-+        println!("[!] libnyx: spawning qemu with:\n {}", params.cmd.join(" "));
-
-         let (shm_work_dir, file_lock) = Self::create_shm_work_dir();
-         let mut shm_work_dir_path = PathBuf::from(&shm_work_dir);
-EOF
+git ls-tree HEAD fuzz_runner/src/nyx/qemu_process.rs
```

Comment:
Do we need `ls-tree` here? It's fine for debugging, but then it's weird to only have this one file and not all patched files.

Reply:
Nope.  This was left over from testing.